### PR TITLE
fix(multi_agent) Deduplicate forked subagent developer instructions

### PR DESCRIPTION
**Pull request by dylan-hurd-oai**
## Summary
- drop inherited forked developer instructions that the child thread will inject again at startup
- add a regression test covering the duplicate-developer-instructions fork path
- confirmed against the referenced rollout bundle that the duplicate exists in the child thread's first raw request payload, so this is not just trace rendering noise

## Testing
- `cargo test -p codex-core spawn_agent_can_fork_parent_thread_history_with_sanitized_items`
- `just fmt`
- `cargo test -p codex-core` *(blocked by a stack overflow in `agent::control::tests::spawn_agent_fork_last_n_turns_keeps_only_recent_turns`; the targeted regression above passes)*

### CODE REVIEW

File: codex-rs/core/src/agent/control.rs [hunk 1]
```diff
@@ -417,11 +416,16 @@ impl AgentControl {
             } else {
                 Vec::new()
             };
+        if let Some(developer_instructions) = config.developer_instructions.as_deref()
```
Comment (jif-oai) [inline codex-rs/core/src/agent/control.rs:419]: We should consider the parent config, not the child config. Not a big deal for now but if their config diverge one day, this will break

**Review by jif-oai — COMMENTED** [review]
This only removes developer messages whose entire `content` is exactly one `InputText`.
But thenormal startup developer message is built multiple stuff. so `developer_instructions` is just one content item alongside permissions/collaboration/skills/plugins/...
